### PR TITLE
Mocha runner changed from tdd to bdd

### PR DESCRIPTION
**Pull request by mabels**
the update of mocha needs a cleanup of the test runner. The tdd
    test runner was used but tdd does not contain beforeEach. Now the
    bdd test runner is use but suite and test are switched to describe
    and it.
    Introduced some validations and test which are used in the TileApi
    feature.

Version Bump includes package.json and package-lock.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,9 +57,9 @@
     "lint": "eslint src/ test/ example/ cli.js",
     "lint:fix": "npm run lint -- --fix",
     "pretest": "npm run lint && npm run build",
-    "unittest": "mocha -u tdd -r babelhook --recursive test/unit/ || true",
-    "unittest:bail": "mocha -u tdd -r babelhook --recursive test/unit/ --bail || true",
-    "test": "istanbul cover -root lib/ node_modules/mocha/bin/_mocha -- -u tdd -r babelhook --recursive test/unit/",
+    "unittest": "mocha -u bdd -r babelhook --recursive test/unit/ || true",
+    "unittest:bail": "mocha -u bdd -r babelhook --recursive test/unit/ --bail || true",
+    "test": "istanbul cover -root lib/ node_modules/mocha/bin/_mocha -- -u bdd -r babelhook --recursive test/unit/",
```
Comment (Sawtaytoes) [inline package.json:62]: What's the purpose of the change from `tdd` to `bdd`?

File: src/lifx/utils.js [hunk 2]
```diff
@@ -125,20 +186,22 @@ utils.rgbHexStringToObject = function(rgbHexString) {
...
+ * @param {Array} array of numbers
+ * @return {Number} minium of the given array
+ */
 utils.minNumberInArray = function(array) {
-  const sortedCopy = array.slice();
-  sortedCopy.sort(function(a, b) {
-    return a - b;
-  });
-  return sortedCopy[0];
+  return Math.min.apply(undefined, array);
```
Comment (Sawtaytoes) [inline src/lifx/utils.js]: Can't we just do `...array` instead or is this a Node.js compatibility thing?
```suggestion
  return Math.min(...array);
```

File: src/lifx/utils.js [hunk 3]
```diff
@@ -125,20 +186,22 @@ utils.rgbHexStringToObject = function(rgbHexString) {
...
+ * @param {Array} array of numbers
+ * @return {Number} maxinum of the given array
+ */
 utils.maxNumberInArray = function(array) {
-  const sortedCopy = array.slice();
-  sortedCopy.sort(function(a, b) {
-    return a - b;
-  });
-  return sortedCopy[sortedCopy.length - 1];
+  return Math.max.apply(undefined, array);
```
Comment (Sawtaytoes) [inline src/lifx/utils.js]: Same comment as with `Math.min`.

File: src/lifx/validate.js [hunk 4]
```diff
@@ -82,8 +82,8 @@ validate.irBrightness = function(brightness, context) {
  * @param {String} context validation context
  */
 validate.optionalKelvin = function(kelvin, context) {
-  const message = 'LIFX %s expects kelvin to be a number between 2500 and 9000';
   if (kelvin !== undefined) {
+    const message = `LIFX %s expects kelvin to be a number between ${constants.HSBK_MINIMUM_KELVIN} and ${constants.HSBK_MAXIMUM_KELVIN}`;
```
Comment (Sawtaytoes) [inline src/lifx/validate.js:86]: I agree with this change, but also, the min and max are different for different bulbs so technically these aren't constants at all. These are constants from the min and max across their entire line of bulbs, but then again, my Day & Night bulbs go down to `1500K` and only up to `4000K`.

Instead of constants, can we pull these values from the information on that specific bulb?

File: src/lifx/validate.js [hunk 5]
```diff
@@ -167,6 +168,54 @@ validate.optionalZoneIndex = function(index, context) {
...
+};
+
+/**
+ * test if the given value is an uint16 value
+ * @param {Number} val the given uint16 value as number
+ * @param {String} context the string for the error message
+ * @return {Boolean} const true or an exception
+ */
+validate.isUInt16 = function(val, context) {
+  return validate.isUIntRange(val, context, 0xffff);
```
Comment (Sawtaytoes) [inline src/lifx/validate.js:208]: Why are we adding these new functions?

File: test/unit/utils-test.js [hunk 6]
```diff
@@ -145,14 +147,94 @@ suite('Utils', () => {
       vendorName: 'LIFX',
       productName: 'White 800 (Low Voltage)',
       productFeatures: {
+        chain: false,
         color: false,
         infrared: false,
-        multizone: false
+        multizone: false,
+        ['temperature_range']: [2700, 6500]
```
Comment (Sawtaytoes) [inline test/unit/utils-test.js:154]: We should probably also add a test for the Day & Night bulbs from 1500 to 4000K.

File: package.json (hunk 1 above)
Comment (mabels) [reply package.json:62]: the tdd specs does not include before_each but we use it.
See commit message.

File: src/lifx/validate.js (hunk 4 above)
Comment (mabels) [reply src/lifx/validate.js:86]: I also so that during the implementation.
I will prepare another pr for this is in the code base.

File: src/lifx/validate.js (hunk 5 above)
Comment (mabels) [reply src/lifx/validate.js:208]: i need them later

File: test/unit/utils-test.js (hunk 6 above)
Comment (mabels) [reply test/unit/utils-test.js:154]: Agreed, in the next pr which cleans the kelvin matter.

File: package.json (hunk 1 above)
Comment (Sawtaytoes) [reply package.json:62]: Understood. This change is fine then.

File: src/lifx/validate.js (hunk 4 above)
Comment (Sawtaytoes) [reply src/lifx/validate.js:86]: Cool. I'll resolve this then.

File: src/lifx/validate.js (hunk 5 above)
Comment (Sawtaytoes) [reply src/lifx/validate.js:208]: 👍
